sdk/endpoint/endpoint.go

```diff
@@ -97,54 +111,71 @@ func (nsme *nsmEndpoint) Start() error {
 		return err
 	}
 
-	// spawn the listnening thread
+	// spawn the listening thread
 	nsme.serve(listener)
 
-	span := spanhelper.FromContext(nsme.Context, fmt.Sprintf("Endpoint-%v-Start", nsme.Configuration.EndpointNetworkService))
-	span.LogObject("labels", nsme.Configuration.EndpointLabels)
+	nsme.registryClient = registry.NewNetworkServiceRegistryClient(nsme.GrpcClient)
+	for i := range nsme.registrations {
+		nsme.register(&nsme.registrations[i])
+	}
+
+	return nil
+}
+
+func (nsme *nsmEndpoint) register(r *registration) {
+	span := spanhelper.FromContext(nsme.Context, fmt.Sprintf("Endpoint-%v-Start", r.Name))
+	span.LogObject("labels", r.Labels)
 	defer span.Finish()
 
 	// Registering NSE API, it will listen for Connection requests from NSM and return information
 	// needed for NSE's forwarder programming.
 	nse := &registry.NetworkServiceEndpoint{
-		NetworkServiceName: nsme.Configuration.EndpointNetworkService,
+		NetworkServiceName: r.Name,
 		Payload:            "IP",
-		Labels:             tools.ParseKVStringToMap(nsme.Configuration.EndpointLabels, ",", "="),
+		Labels:             r.Labels,
 	}
 	registration := &registry.NSERegistration{
 		NetworkService: &registry.NetworkService{
-			Name:    nsme.Configuration.EndpointNetworkService,
+			Name:    r.Name,
 			Payload: "IP",
 		},
 		NetworkServiceEndpoint: nse,
 	}
 	span.LogObject("nse-request", registration)
 
-	nsme.registryClient = registry.NewNetworkServiceRegistryClient(nsme.GrpcClient)
 	registeredNSE, err := nsme.registryClient.RegisterNSE(span.Context(), registration)
 	if err != nil {
 		span.Logger().Fatalln("unable to register endpoint", err)
 	}
-	nsme.endpointName = registeredNSE.GetNetworkServiceEndpoint().GetName()
-	span.LogObject("endpoint-name", nsme.endpointName)
+	r.registeredName = registeredNSE.GetNetworkServiceEndpoint().GetName()
+	span.LogObject("endpoint-name", r.registeredName)
 	span.Logger().Infof("NSE registered: %v", registeredNSE)
 	span.Logger().Infof("NSE: channel has been successfully advertised, waiting for connection from NSM...")
-
-	return nil
 }
 
-func (nsme *nsmEndpoint) Delete() error {
-	span := spanhelper.FromContext(context.Background(), fmt.Sprintf("Endpoint-%v-Delete", nsme.Configuration.EndpointNetworkService))
+func (nsme *nsmEndpoint) deregister(r *registration) error {
```

Comment:
[minor]
```suggestion
func (nsme *nsmEndpoint) unregister(r *registration) error {
```

Reply:
It can add some consistent with the registry API https://github.com/networkservicemesh/api/blob/master/pkg/api/registry/registry.proto#L40